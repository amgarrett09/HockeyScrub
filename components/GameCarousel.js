import React from 'react';
import {
  CarouselProvider,
  Slider,
  Slide,
  ButtonBack,
  ButtonNext,
} from 'pure-react-carousel';
import PropTypes from 'prop-types';
import GameInfo from './GameInfo';
import 'pure-react-carousel/dist/react-carousel.es.css';
import '../css/game-carousel.css';

const GameCarousel = ({ games }) => (
  <CarouselProvider
    naturalSlideWidth={175}
    naturalSlideHeight={100}
    totalSlides={games.length}
    visibleSlides={games.length > 4 ? 4 : games.length}
    className="game-carousel"
  >
    <ButtonBack className="game-carousel__button">
      <strong>&lt;</strong>
    </ButtonBack>
    <Slider className="game-carousel__slider">
      {games.map((game, i) => (
        <Slide
          key={game.awayTeam + game.homeTeam}
          index={i}
          data-test="slide"
        >
          <GameInfo
            awayTeam={game.awayTeam}
            awayScore={game.awayScore}
            homeTeam={game.homeTeam}
            homeScore={game.homeScore}
            gameState={game.gameState}
            stateCode={game.stateCode}
            gameTime={game.gameTime}
          />
        </Slide>
      ))}
    </Slider>
    <ButtonNext className="game-carousel__button">
      <strong>&gt;</strong>
    </ButtonNext>
  </CarouselProvider>
);

GameCarousel.propTypes = {
  games: PropTypes.arrayOf(
    PropTypes.shape({
      awayTeam: PropTypes.string,
      awayScore: PropTypes.number,
      homeTeam: PropTypes.string,
      homeScore: PropTypes.number,
      gameState: PropTypes.string,
    }),
  ).isRequired,
};

export default GameCarousel;
