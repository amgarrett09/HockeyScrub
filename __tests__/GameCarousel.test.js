/* eslint-env jest */

import { shallow } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';
import GameCarousel from '../components/GameCarousel';

const games = [
  {
    awayTeam: 'Detroit Red Wings',
    awayScore: 5,
    homeTeam: 'Buffalo Sabres',
    homeScore: 4,
    gameState: 'FINAL',
  },
  {
    awayTeam: 'Chicago Blackhawks',
    awayScore: 3,
    homeTeam: 'Winnipeg Jets',
    homeScore: 4,
    gameState: '3RD',
  },
  {
    awayTeam: 'San Jose Sharks',
    awayScore: 2,
    homeTeam: 'Arizona Coyotes',
    homeScore: 1,
    gameState: '1ST',
  },
];

describe('GameCarousel', () => {
  const component = <GameCarousel games={games} />;

  it('should have three slides in the slider', () => {
    const wrapper = shallow(component);
    const slides = wrapper.find('[data-test="slide"]');

    expect(slides.length).toEqual(3);
  });

  it('sould render and should not change unexpectedly', () => {
    const wrapper = renderer.create(component);
    const tree = wrapper.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
