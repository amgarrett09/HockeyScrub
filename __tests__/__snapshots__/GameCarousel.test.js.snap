// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameCarousel sould render and should not change unexpectedly 1`] = `
<div
  className="carousel game-carousel"
>
  <button
    aria-label="previous"
    className="buttonBack___1mlaL carousel__back-button game-carousel__button"
    disabled={true}
    onClick={[Function]}
    type="button"
  >
    <strong>
      &lt;
    </strong>
  </button>
  <div
    aria-live="polite"
    className="horizontalSlider___281Ls carousel__slider carousel__slider--horizontal game-carousel__slider"
    onKeyDown={[Function]}
    role="listbox"
    style={Object {}}
    tabIndex={0}
  >
    <div
      className="carousel__slider-tray-wrapper carousel__slider-tray-wrap--horizontal"
      style={Object {}}
    >
      <ul
        className="sliderTray___-vHFQ sliderAnimation___300FY carousel__slider-tray carousel__slider-tray--horizontal"
        onClickCapture={[Function]}
        onMouseDown={[Function]}
        onTouchCancel={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "transform": "translateX(0%) translateX(0px)",
            "width": "100%",
          }
        }
      >
        <li
          aria-selected={true}
          className="slide___3-Nqo slideHorizontal___1NzNV carousel__slide carousel__slide--visible"
          data-test="slide"
          onBlur={[Function]}
          onFocus={[Function]}
          role="option"
          style={
            Object {
              "paddingBottom": "19.047619047619047%",
              "width": "33.333333333333336%",
            }
          }
          tabIndex={0}
        >
          <div
            className="slideInner___2mfX9 carousel__inner-slide"
          >
            <div
              className="game-info"
            >
              <div
                className="game-info__scores"
              >
                <div
                  className="row"
                >
                  <div
                    className="col s6"
                    data-test="away-team-name"
                  >
                    DET
                  </div>
                  <div
                    className="col s6"
                    data-test="away-team-score"
                  >
                    5
                  </div>
                </div>
                <div
                  className="row"
                >
                  <div
                    className="col s6"
                    data-test="home-team-name"
                  >
                    BUF
                  </div>
                  <div
                    className="col s6"
                    data-test="home-team-score"
                  >
                    4
                  </div>
                </div>
              </div>
              <div
                className="game-info__game-state"
                data-test="game-status"
              >
                FINAL
              </div>
            </div>
          </div>
        </li>
        <li
          aria-selected={true}
          className="slide___3-Nqo slideHorizontal___1NzNV carousel__slide carousel__slide--visible"
          data-test="slide"
          onBlur={[Function]}
          onFocus={[Function]}
          role="option"
          style={
            Object {
              "paddingBottom": "19.047619047619047%",
              "width": "33.333333333333336%",
            }
          }
          tabIndex={0}
        >
          <div
            className="slideInner___2mfX9 carousel__inner-slide"
          >
            <div
              className="game-info"
            >
              <div
                className="game-info__scores"
              >
                <div
                  className="row"
                >
                  <div
                    className="col s6"
                    data-test="away-team-name"
                  >
                    CHI
                  </div>
                  <div
                    className="col s6"
                    data-test="away-team-score"
                  >
                    3
                  </div>
                </div>
                <div
                  className="row"
                >
                  <div
                    className="col s6"
                    data-test="home-team-name"
                  >
                    WIN
                  </div>
                  <div
                    className="col s6"
                    data-test="home-team-score"
                  >
                    4
                  </div>
                </div>
              </div>
              <div
                className="game-info__game-state"
                data-test="game-status"
              >
                3RD
              </div>
            </div>
          </div>
        </li>
        <li
          aria-selected={true}
          className="slide___3-Nqo slideHorizontal___1NzNV carousel__slide carousel__slide--visible"
          data-test="slide"
          onBlur={[Function]}
          onFocus={[Function]}
          role="option"
          style={
            Object {
              "paddingBottom": "19.047619047619047%",
              "width": "33.333333333333336%",
            }
          }
          tabIndex={0}
        >
          <div
            className="slideInner___2mfX9 carousel__inner-slide"
          >
            <div
              className="game-info"
            >
              <div
                className="game-info__scores"
              >
                <div
                  className="row"
                >
                  <div
                    className="col s6"
                    data-test="away-team-name"
                  >
                    SJS
                  </div>
                  <div
                    className="col s6"
                    data-test="away-team-score"
                  >
                    2
                  </div>
                </div>
                <div
                  className="row"
                >
                  <div
                    className="col s6"
                    data-test="home-team-name"
                  >
                    ARI
                  </div>
                  <div
                    className="col s6"
                    data-test="home-team-score"
                  >
                    1
                  </div>
                </div>
              </div>
              <div
                className="game-info__game-state"
                data-test="game-status"
              >
                1ST
              </div>
            </div>
          </div>
        </li>
      </ul>
    </div>
  </div>
  <button
    aria-label="next"
    className="buttonNext___2mOCa carousel__next-button game-carousel__button"
    disabled={true}
    onClick={[Function]}
    type="button"
  >
    <strong>
      &gt;
    </strong>
  </button>
</div>
`;
